File: answers/Fab-Gaurav/Day 17/Answer_2.cpp
#include <bits/stdc++.h>
using namespace std;

/*
void subArr(int arr[],int i,int j)
{
    for(int k=i;k<=j;k++)
    cout<<arr[k]<<" ";
    cout<<'\n';
}
*/

int main()
{
    int n, arr[20], sum=0, cnt=0;
    cout<<"Enter the number of elements in array : ";
    cin>>n;
    cout<<"Enter the elements of array : ";
    for(int i=0; i<n; i++)
    {
        cin>>arr[i];
    }
    for(int i=0; i<n; i++)
    {
        sum=0;
        for(int j=i; j<n; j++)
        {
            sum=arr[j]+sum;
            if(sum==0)
            {
                cnt++;
                /*
                subArr(arr,i,j);              //This function will display the sub arrays
                */
            }
        }
    }
    cout<<"Total sub arrays are : "<<cnt;
    
    if(cnt==0)
        cout<<"No such array";
    return 0;
}

File: answers/Fab-Gaurav/Day 14/Answer_2.py
def add(arr_1 ,arr_2,i=0):
    if i<len(arr_1 ):
        return [arr_1 [i] + arr_2[i]] + add(arr_1 ,arr_2,i+1)
    return[]

arr_1 =[int(x) for x in input('Enter the elements of array 1 : ').split()]
arr_2=[int(x) for x in input('Enter the elements of array 2 : ').split()]

l=len(arr_1 )-len(arr_2)
if l>0:
    arr_2+=[0]*l
else:
    arr_1 +=[0]*abs(l)
print('Sum of the arrays : ',add(arr_1 ,arr_2))